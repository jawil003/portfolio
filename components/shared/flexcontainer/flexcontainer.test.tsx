import { render } from "@testing-library/react";
import {
  FlexContainer,
  FlexContainerProps,
} from ".";

const variants: FlexContainerProps[] = [
  {
    alignContent: "flex-start",
    justifyContent: "flex-start",
    alignItems: "flex-start",
    direction: "row",
  },
  {
    columnGap: "10px",
    rowGap: "10px",
  },
  {
    wrap: true,
  },
];

describe("FlexContainer", () => {
  test.each(variants)(
    "FlexContainer with %p",
    (variant) => {
      const testFlexContainer = render(
        <FlexContainer {...variant} />,
      );

      expect(
        testFlexContainer.container
          .firstChild,
      ).toMatchSnapshot();
    },
  );
});
