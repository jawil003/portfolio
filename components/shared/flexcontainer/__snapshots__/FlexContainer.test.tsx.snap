// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexContainer FlexContainer with {"alignContent": "flex-start", "alignItems": "flex-start", "direction": "row", "justifyContent": "flex-start"} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0px / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0px / 2) calc(0px / 2);
}

<div
  class="emotion-0"
/>
`;

exports[`FlexContainer FlexContainer with {"columnGap": "10px", "rowGap": "10px"} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(10px / -2) calc(10px / -2);
}

.emotion-0>* {
  margin: calc(10px / 2) calc(10px / 2);
}

<div
  class="emotion-0"
/>
`;

exports[`FlexContainer FlexContainer with {"wrap": true} 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: calc(0px / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0px / 2) calc(0px / 2);
}

<div
  class="emotion-0"
/>
`;
